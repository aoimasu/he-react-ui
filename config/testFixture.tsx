import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
