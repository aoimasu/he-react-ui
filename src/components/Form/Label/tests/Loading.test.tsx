import fixture from '../fixtures/Loading.fixture';
import testFixture from '../../../../../config/testFixture';

testFixture(fixture, 'Loading');
