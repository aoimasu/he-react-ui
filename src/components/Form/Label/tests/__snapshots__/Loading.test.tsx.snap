// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> rendered correctly with Loading fixture 1`] = `
<Label
  error={false}
  htmlFor="demo"
>
  <div
    className="label"
  >
    <LoadingStrip
      className="loading"
    >
      <div
        className="outer loading"
      >
        <div
          className="inner"
        />
      </div>
    </LoadingStrip>
  </div>
</Label>
`;
